main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java

```diff
@@ -183,10 +193,12 @@ private void loadRouteBasedFaresComponent(GtfsFeedContainer feedContainer) {
                 List.of(
                     GtfsFareLegRule::hasFromAreaId,
                     (Function<GtfsFareLegRule, Boolean>) GtfsFareLegRule::hasToAreaId))
-            && hasAtLeastOneRecordForFields(
-                feedContainer,
-                GtfsRoute.FILENAME,
-                List.of((Function<GtfsRoute, Boolean>) GtfsRoute::hasNetworkId)));
+            && (hasAtLeastOneRecordForFields(
```

Comment:
Is this the right rules for Route-Based Fares? According to the issue, it should be:
`Add Route-Based Fares = If at least 1 value of network_id in routes.txt defined OR 1 record in networks.txt`
I am not sure it was right even _before_ the change.

Reply:
It does look like we were checking the logic differently from this before. We don't need to check for fare_leg_rule now, just network_id.routes.txt OR networks.txt

Reply:
Done.